### DISCUSSION TITLE
Mini-RFC: Remove `TestStore` scoping

### DISCUSSION THREAD

**Discussion by stephencelis** [pointfreeco/swift-composable-architecture#2291] [Request for comment]
We are revisiting an old RFC, and are planning to remove the ability for test stores to be scoped to view state and actions: https://github.com/pointfreeco/swift-composable-architecture/discussions/1148

We believe this feature is cumbersome enough to not be widely used, and we believe there are probably better solutions to testing view state and actions. If anyone _is_ using test store scoping, please let us know and we'd love to help brainstorm other approaches to the problem.

**Comment by 3a4oT** [comment]
Hey, I'm confused by this change. Now I have  ~600 warnings on tests, and I'm trying to figure out how to resolve those.

`'init(initialState:reducer:prepareDependencies:file:line:)' is deprecated: Test the reducer domain directly. To test view state and actions, write a unit test.`  Make no sense to me since I'm already in unit tests.

**Reply by mbrandonw** [reply]
Hi @3a4oT, sorry about that. We had an incorrect deprecation message in the last release that has been fixed on main, but not yet released as a point release.

Test stores now take their reducer as a builder closure. So, to fix your deprecation warnings you need to change any test store that looks like this:

```swift
let store = TestStore(initialState: Feature.State(), reducer: Feature())
```

…to this:

```swift
let store = TestStore(initialState: Feature.State()) {
  Feature()
}
```

Or if you can construct a fancy search-and-replace, you can turn it into this:

```swift
let store = TestStore(initialState: Feature.State(), reducer: { Feature() })
```

Sorry for the inconvenience, but on the plus side, congrats on having such an extensive test suite! 😁

**Comment by 3a4oT** [comment]
Thanks for the reply. Now it's clear.
